lectures/Lecture3.3_ConvNets.ipynb

```diff
@@ -0,0 +1,2382 @@
+    "\n",
+    "We would calculate: \n",
+    "\n",
+    "- $ WX = A = \\displaystyle \\begin{bmatrix}\n",
+    "                            a_{11} & a_{12} & a_{13} & ... &  a_{1m} \\\\\n",
+    "                            a_{21} & a_{22} & a_{23} & ... &  a_{2m} \\\\\n",
+    "                            a_{31} & a_{32} & a_{33} & ... &  a_{3m} \\\\\n",
+    "                            \\vdots & \\vdots & \\vdots & \\ddots & \\vdots \\\\\n",
+    "                            a_{m1} & a_{m2} & a_{m3} & ... & a_{mm} \n",
+    "                            \\end{bmatrix}    $  \n",
+    "\n",
+    "\n",
+    "Each element of $ A $ would be the dot product (weighted sum) between one row of $ W $ and one column of $ X $: \n",
+    "\n",
+    "\n",
+    "$ \\displaystyle a_{ij} = \\sum_{j = 1}^{n} w_{ij}x_{ji} = w_{i1}x_{1j} + w_{i2}x_{2j} + w_{i3}x_{3j} + ... + w_{in}x_{nj} $ \n",
+    "\n",
+    "where \n",
+    "\n",
+    "- $ a_{ij} = $ is the response of the $ i^{th} $ neuron to the $ j^{th} $ column of the input matrix \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "While feasible in principle, this would be:\n",
+    "\n",
+    "- Quite complex  \n",
+    "  - If you tried it out with paper and pencil, you might have a hard time wrapping your mind around it\n",
+    "\n",
+    "- Lacking a straightforward interpretation  \n",
+    "\n",
+    "- Very inefficient\n",
+    "    - It neglects the compositional structure of inputs\n",
+    "    \n",
+    "      - Wait - what is this?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### **2.2. Convolution Captures Local Structures** <a class=\"anchor\" id=\"22-convolution-captures-local-structures\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Often, grid-like inputs are a composition of local structures, or _**features**_:\n",
+    "\n",
+    "- An image is composed of edges, circles, and other geometric primitives\n",
+    "- An EEG is composed of different patterns of activity from different electrodes\n",
+    "- etc.\n",
+    "\n",
+    "To capture this local structure, we can:\n",
+    "\n",
+    "- Take a small weights matrix (smaller than the input) \n",
+    "- Overlay this small weights matrix to an equally small patch of input\n",
+    "- Multiply the small weights matrix and the small patch of input element-wise\n",
+    "- Sum all the element-wise products\n",
+    "- Repeat on a different patch until the whole input is covered\n",
+    "\n",
+    "This operation is known as _**(discrete) convolution**_ and is equivalent to taking the dot product between a patch of input and a set of weights."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that:\n",
+    "\n",
+    "- Convolution returns weighted sums of inputs that are specific to a given input region, that is, that represent a specific _feature_\n",
```

Comment:
The video is good but I feel that a video where one see (graphically) how matching kernel and underlying values result in high values for that entry of the convolution would make the explanation stronger for who isn't familiar with matrices

Reply:
I am sorry, I don't understand what you mean :/